lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepository.java

```diff
@@ -42,27 +42,25 @@
 public class LightblueNotificationRepository implements NotificationRepository {
     private final LightblueClient lightblue;
     private final String[] entities;
-    private final Locking locking;
+    private final LockStrategy lockStrategy;
     private final Map<String, NotificationFactory> notificationFactoryByEntityName;
     private final Clock clock;
 
     public LightblueNotificationRepository(LightblueClient lightblue, String[] entities,
-            String lockingDomain, Map<String, NotificationFactory> notificationFactoryByEntityName,
-            Clock clock) {
+            LockStrategy lockStrategy,
+            Map<String, NotificationFactory> notificationFactoryByEntityName, Clock clock) {
         this.lightblue = lightblue;
         this.entities = entities;
         this.notificationFactoryByEntityName = notificationFactoryByEntityName;
         this.clock = clock;
-
-        locking = lightblue.getLocking(lockingDomain);
+        this.lockStrategy = lockStrategy;
     }
 
     @Override
     public List<LightblueNotification> retrieveOldestNotificationsUpTo(int maxEvents)
-            throws LightblueException {
-        try {
-            blockUntilLockAcquired(Locks.forNotificationsForEntities(entities));
-
+            throws Exception {
+        try (LockedResource lock = lockStrategy
+                .blockUntilAcquired(ResourceIds.forNotificationsForEntities(entities))) {
             BulkLightblueRequester requester = new BulkLightblueRequester(lightblue);
```

Comment:
One more thing to consider re: locking by entity name: if lock owner dies, all entity even processing stops until lock expires.